astropy/units/core.py

```diff
@@ -2587,34 +2587,35 @@ def def_unit(
 
 
 def _condition_arg(value):
-    """
-    Validate value is acceptable for conversion purposes.
+    """Validate value is acceptable for conversion purposes.
 
-    Will convert into an array if not a scalar, and can be converted
-    into an array
+    Will convert into an array if not a scalar or array-like, where scalars
+    and arrays can be python and numpy types, anything that defines
+    ``__array_namespace__`` or anything that has a ``.dtype`` attribute.
 
     Parameters
     ----------
-    value : int or float value, or sequence of such values
+    value : scalar or array-like
 
     Returns
     -------
-    Scalar value or numpy array
+    Scalar value or array
 
     Raises
     ------
     ValueError
         If value is not as expected
+
     """
-    if isinstance(value, (np.ndarray, float, int, complex, np.void)):
+    if (
+        isinstance(value, (np.ndarray, float, int, complex))
```

Comment:
Pre-construct the tuple?

Reply:
Makes sense - though I now remember I just found out last week that it is suggested to write this in a more typing-ready form (which I checked is equally fast),
```
KNOWN_GOOD = np.ndarray | float | int | complex
```
Let me go with that here too.